File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        String xAsStr = String.valueOf(x);
        int length = xAsStr.length();
        for (int i = 0; i<length/2; i++) {
            if (xAsStr.charAt(i) != xAsStr.charAt(length-1-i)) {
                return false;
            }
        }
        return true;
    }
}

File: 0009-palindrome-number/0009-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        string xAsStr = to_string(x);
        for (int i=0; i<xAsStr.length()/2; i++) {
           if (xAsStr[i] != xAsStr[xAsStr.length()-i-1]) {
                return false;
            }
        }
        return true;
    }
};